python/cudf/cudf/_lib/pylibcudf/column_factories.pyx

```diff
@@ -0,0 +1,266 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+from libcpp.memory cimport unique_ptr
+from libcpp.utility cimport move
+
+from cudf._lib.cpp.column.column cimport column
+from cudf._lib.cpp.column.column_factories cimport (
+    make_duration_column as cpp_make_duration_column,
```

Comment:
I'm confused. Why are these in `cudf._lib` and not `pylibcudf`? Are we creating high-level APIs in `pylibcudf` before we migrate the lower-level APIs from `cudf._lib` to `pylibcudf`? That seems like an odd order of operations.

Reply:
There isn't a clean, one-size-fits-all approach to migrate cudf._lib.cpp bits into pylibcudf because of how many interdependencies there are between these files. Unless what you're envisioning is a wholesale move of that entire directory into pylibcudf, which could be done with a fairly straightforward find-and-replace but would be a massive PR.

Reply:
It seems like we'd want to make a clean break so that cudf doesn't directly interface with libcudf but only does so via pylibcudf, and then all other work is scoped under "clean up pylibcudf."

Reply:
This is done in https://github.com/rapidsai/cudf/pull/15740.